src/Form/Form.js

```diff
@@ -442,6 +443,30 @@ class Form {
                 return
             }
         }
+
+        // if we have an unknown input, and only one non-hidden password input, we can assume the unknown input is the username
+        // In that case we move the inputs to the correct category
+        if (this.canCategorizeUnknownUsername()) {
+            const hasUsername = [...this.inputs.all].some(input => getInputSubtype(input) === 'username')
```

Comment:
If it's a `username` it's definitely in the `inputs.credentials` set, which is always smaller, potentially much smaller. 🙂

Reply:
Since we're there, you're destructuring and looping again a few lines below. Maybe only destructure once and/or use `for…of`.